### ISSUE TITLE
flatten input_shape does not accept mode.output.shape[1:]

### ISSUE THREAD

**Issue by ptisseur** [fchollet/keras#6125]
I am trying to modify the second code in the blog Building powerful image classification models using very little data.
The aim is to build a 8 classes classifier using the pretrain VGG16 model.
I have a train_data.shape=train_data.shape[1:]=(3392,4,4,512)
Trying to create the top part of the model with 
model = Sequential()
    model.add(Flatten(input_shape=train_data.shape[1:])) #  1 case
    #model.add(Flatten(input_shape=(3392,4,4,512)))  #2 case
    model.add(Flatten(input_shape=(3392,4,4,512)))
    model.add(Dense(256, activation='relu'))
    model.add(Dropout(0.5))
    model.add(Dense(8, activation='softmax'))
In case 1 I obtain expexted flatten1 to have (None,8192)  but got array (3401,1)
In case 2
i obtain valueError expexted flatten1 input to have 5 dimensions but got array (3392,4,4,512)
I habe the 2.02 version and tensoflow as backend.

The (None,8192) looks like the output of the flatten layen and not the input.
I dont know what to do 
Thanks 

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
